File: app/src/main/java/com/anadi/weatherinfo/addlocation/LocationsProvider.java
package com.anadi.weatherinfo.addlocation;

import android.content.Context;

import com.anadi.weatherinfo.repository.Country;

import java.util.ArrayList;

public interface LocationsProvider {
    void setContext(Context context);
    void loadLocations();

    ArrayList<String> getCityNames(String country);
    ArrayList<String> getCountryNames();

    Country getCountryByName(String countryName);

    Country getRandomCountry();
    String getRandomCity(String countryName);
}

File: app/src/main/java/com/anadi/weatherinfo/addlocation/LocationsProvider.kt
package com.anadi.weatherinfo.addlocation

import android.content.Context
import com.anadi.weatherinfo.repository.Country
import java.util.*

interface LocationsProvider {
    fun loadLocations(context: Context?)
    fun getCityNames(country: String?): ArrayList<String?>?
    val countryNames: ArrayList<String?>?
    fun getCountryByName(countryName: String?): Country?
    val randomCountry: Country?
    fun getRandomCity(countryName: String?): String?
}